### PR TITLE
Added 2 new linked list operations one for finding cycles and one for reversing the list

### PR DESCRIPTION
**Pull request by achak47**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: data_structures/linked_list/detecting_cycle_in_list.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#include <stdio.h>
```
Comment (alexpantyukhin) [inline data_structures/linked_list/detecting_cycle_in_list.c:1]: Please add a one-line description of what the library/header is for

File: data_structures/linked_list/detecting_cycle_in_list.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		if (slow_p == fast_p) {
+			return 1;
+		}
+	}
+	return 0;
+}
+
+int main()
+{
+	struct Node* head = NULL;
```
Comment (alexpantyukhin) [inline data_structures/linked_list/detecting_cycle_in_list.c:37]: According the https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#test . What do you think about moving your example into the `test` method with asserts?

File: data_structures/linked_list/detecting_cycle_in_list.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	new_node->data = new_data;
+
+	new_node->next = (*head_ref);
+
+	(*head_ref) = new_node;
+}
+
+int detectLoop(struct Node* list)
+{
+	struct Node *slow_p = list, *fast_p = list;
```
Comment (alexpantyukhin) [inline data_structures/linked_list/detecting_cycle_in_list.c:23]: If I remember correctly, It's a Floyd's tortoise and hare algorithm. Probably it's better to add a comment about this. That would be easy to recognize it in the future.

**Review by alexpantyukhin — CHANGES_REQUESTED** [review]

**Comment by Panquesito7** [comment]
Great comments, @alexpantyukhin. Thanks for your help! 🚀

**Pull request closed without merging**